### PR TITLE
chore(query): migrate agg (count, sum, avg) and combinators into v2

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

chore(query): migrate agg (count, sum, avg) and combinators into v2

Fixes #issue

### CODE REVIEW

File: src/common/hashtable/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ test = false
 common-base = { path = "../base" }
 
 # Crates.io dependencies
-ordered-float = "3.0.0"
+ordered-float = { git = "https://github.com/andylokandy/rust-ordered-float.git", branch = "as", features = ["serde"] }
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply src/common/hashtable/Cargo.toml:18]: Let's do it in another pr.

File: src/query/functions-v2/src/aggregates/adaptors/aggregate_null_variadic_adaptor.rs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+use common_expression::Column;
+use common_expression::ColumnBuilder;
+use common_io::prelude::BinaryWriteBuf;
+
+use crate::aggregates::AggregateFunction;
+use crate::aggregates::AggregateFunctionRef;
+use crate::aggregates::StateAddr;
+
+#[derive(Clone)]
+pub struct AggregateNullVariadicAdaptor<const NULLABLE_RESULT: bool, const STKIP_NULL: bool> {
```
Comment (RinChanNOWWW) [inline src/query/functions-v2/src/aggregates/adaptors/aggregate_null_variadic_adaptor.rs:33]: Is `SKIP_NULL`  used? (and typo)

File: src/query/functions-v2/src/aggregates/adaptors/mod.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Datafuse Labs.
+// Copyright 2021 Datafuse Labs.
```
Comment (b41sh) [inline src/query/functions-v2/src/aggregates/adaptors/mod.rs:1]: 2022?

File: src/query/functions-v2/src/aggregates/adaptors/aggregate_null_variadic_adaptor.rs [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+use common_expression::Column;
+use common_expression::ColumnBuilder;
+use common_io::prelude::BinaryWriteBuf;
+
+use crate::aggregates::AggregateFunction;
+use crate::aggregates::AggregateFunctionRef;
+use crate::aggregates::StateAddr;
+
+#[derive(Clone)]
+pub struct AggregateNullVariadicAdaptor<const NULLABLE_RESULT: bool, const SKIP_NULL: bool> {
```
Comment (RinChanNOWWW) [inline src/query/functions-v2/src/aggregates/adaptors/aggregate_null_variadic_adaptor.rs:33]: `SKIP_NULL` seems not used.

File: src/query/functions-v2/src/aggregates/aggregate_combinator_distinct.rs [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                    nested_name: nested_name.to_owned(),
+                    arguments,
+                    nested,
+                    name,
+                    _s: PhantomData,
+                    _state: PhantomData,
+                })),
+            _ =>
+                return Err(ErrorCode::BadDataValueType(format!(
+                    "AggregateSumFunction does not support type '{:?}'",
```
Comment (RinChanNOWWW) [inline src/query/functions-v2/src/aggregates/aggregate_combinator_distinct.rs:237]: AggregateSumFunction？

**Review by RinChanNOWWW — APPROVED** [review]
LGTM

**Review by b41sh — APPROVED** [review]

**Pull request merged**